dhall-docs/src/Dhall/Docs.hs

```diff
@@ -117,202 +100,32 @@ parserInfoOptions =
-    -> String          -- ^ Package name
-    -> IO ()
-createIndexes outputPath htmlFiles packageName = do
-    let toMap file = Map.singleton (Path.parent file) [file]
-    let filesGroupedByDir = Map.unionsWith (<>) $ map toMap htmlFiles
-
-    let listDirRel dir = do
-            dirs <- fst <$> Path.IO.listDir dir
-            mapM (Path.stripProperPrefix dir) dirs
-
-    let createIndex index files = do
-            indexFile <- Path.fromAbsFile . (index </>) <$> Path.parseRelFile "index.html"
-            indexTitle <-
-                if outputPath == index then return "package"
-                else Path.fromRelDir <$> Path.stripProperPrefix outputPath index
-            indexList <- Control.Monad.forM files $
-                fmap Path.filename . Path.stripProperPrefix outputPath
-            dirList <- listDirRel index
-
-            let relativeResourcesPath = resolveRelativePath outputPath index
-            Lucid.renderToFile indexFile $
-                indexToHtml
-                    indexTitle
-                    indexList
-                    dirList
-                    DocParams
-                        { relativeResourcesPath = relativeResourcesPath
-                        , packageName = packageName}
-
-    _ <- Map.traverseWithKey createIndex filesGroupedByDir
-    return ()
-
 -- | Default execution of @dhall-docs@ command
 defaultMain :: Options -> IO ()
 defaultMain = \case
     Options{..} -> do
         GHC.IO.Encoding.setLocaleEncoding System.IO.utf8
         resolvedPackageDir <- Path.IO.resolveDir' packageDir
-        resolvedOutDir <- Path.IO.resolveDir' outDir
-
-        let packageName = packageNameResolver resolvedPackageDir
 
-        dhallFilesAndHeaders <- getAllDhallFilesAndHeaders resolvedPackageDir
-        if null dhallFilesAndHeaders then
-            putStrLn $
-                "No documentation was generated because no file with .dhall " <>
-                "extension was found"
-        else do
-            generatedHtmlFiles <-
-                mapM (uncurry $ saveHtml resolvedPackageDir resolvedOutDir packageName)
-                    dhallFilesAndHeaders
-            createIndexes resolvedOutDir generatedHtmlFiles packageName
-
-            dataDir <- getDataDir
-            Path.IO.ensureDir resolvedOutDir
-            Control.Monad.forM_ dataDir $ \(filename, contents) -> do
-                let finalPath = Path.fromAbsFile $ resolvedOutDir </> filename
-                Data.ByteString.writeFile finalPath contents
+        outDirExists <- System.Directory.doesPathExist docLink
+        Control.Monad.when outDirExists $ do
+            isLink <- System.Directory.pathIsSymbolicLink docLink
+            if isLink then System.Directory.removeFile docLink
+            else die $ "The specified --output-link (" <> docLink <> ") already "
+                    <> "exists and its not a symlink. The --output-link flag "
+                    <> "needs to either not exist or to be a symlink"
```

Comment:
The wording regarding the existence of the flag is a bit awkward. Maybe just shorten it?!

```suggestion
            else die $ "The specified --output-link (" <> docLink <> ") already "
                    <> "exists and it's not a symlink."
```